app/src/main/java/org/wikipedia/yearinreview/YearInReviewViewModel.kt

```diff
@@ -101,16 +106,19 @@ class YearInReviewViewModel() : ViewModel() {
                         .take(MAX_TOP_CATEGORY)
                 }
 
-                val impactDataJob = async {
-                    if (AccountUtil.isLoggedIn) {
-                        val wikiSite = WikipediaApp.instance.wikiSite
+                var totalPageViews = 0L
+                var editCount = 0
+                if (AccountUtil.isLoggedIn) {
+                    val wikiSite = WikipediaApp.instance.wikiSite
+                    val userInfoResponse = ServiceFactory.get(wikiSite).getLocalAndGlobalUserInfo()
+
+                    val impactDataJob = async {
                         val now = Instant.now().epochSecond
                         val impact: GrowthUserImpact
                         val impactLastResponseBodyMap = Prefs.impactLastResponseBody.toMutableMap()
                         val impactResponse = impactLastResponseBodyMap[wikiSite.languageCode]
                         if (impactResponse.isNullOrEmpty() || abs(now - Prefs.impactLastQueryTime) > TimeUnit.HOURS.toSeconds(12)) {
-                            val userId =
-                                ServiceFactory.get(wikiSite).getUserInfo().query?.userInfo?.id!!
+                            val userId = userInfoResponse.query?.userInfo?.id!!
```

Comment:
Since you updated this, do you think it would be better if we update it to `val userId = userInfoResponse.query?.userInfo?.id ?: 0`, just like line `158`?

Reply:
sure!